diesel/src/type_impls/primitives.rs

```diff
@@ -160,25 +147,14 @@ where
     DB: Backend,
     *const [u8]: FromSql<ST, DB>,
 {
-    fn from_sql(bytes: Option<&DB::RawValue>) -> deserialize::Result<Self> {
+    fn from_sql<'a>(bytes: Option<<<DB as Backend>::RawValue as FamilyLt<'a>>::Out>) -> deserialize::Result<Self> {
```

Comment:
```suggestion
    fn from_sql<'a>(bytes: Option<RawValue<'a, DB>>) -> deserialize::Result<Self> {
```

Reply:
This doesn't seem to be a valid change because the expected type is:
fn(std::option::Option<<::RawValue as backend::FamilyLt<'a>>::Out>)